sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java

```diff
@@ -91,6 +90,10 @@ private static String getIdFromLine(String line) {
     int startIdx = lastSection.lastIndexOf('-');
     int endIdx = lastSection.lastIndexOf('.');
 
+    if (startIdx > endIdx && endIdx != -1) {
+      return null;
+    }
+
     String containerId =
         lastSection.substring(
             startIdx == -1 ? 0 : startIdx + 1, endIdx == -1 ? lastSection.length() : endIdx);
```

Comment:
```suggestion
    if (startIdx == -1) {
      startIdx = 0;
    }
    if (endIdx == -1) {
      endIdx = lastSection.length();
    }
    if (startIdx > endIdx) {
      return null;
    }

    String containerId = lastSection.substring(startIdx, endIdx);
```

How about let's simplify this a bit